libfwupdplugin/fu-self-test.c

```diff
@@ -2308,6 +2308,77 @@ fu_firmware_dfu_func(void)
 	g_assert_true(ret);
 }
 
+static void
+fu_firmware_ifwi_cpd_func(void)
+{
+	gboolean ret;
+	g_autofree gchar *filename_ifwi_cpd = NULL;
+	g_autoptr(FuFirmware) firmware = fu_ifwi_cpd_firmware_new();
+	g_autoptr(FuFirmware) img1 = NULL;
+	g_autoptr(FuFirmware) img2 = NULL;
+	g_autoptr(GBytes) data_bin = NULL;
+	g_autoptr(GBytes) data_ifwi_cpd = NULL;
+	g_autoptr(GError) error = NULL;
+
+	filename_ifwi_cpd = g_test_build_filename(G_TEST_DIST, "tests", "ifwi-cpd.bin", NULL);
+	data_ifwi_cpd = fu_bytes_get_contents(filename_ifwi_cpd, &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(data_ifwi_cpd);
+	ret = fu_firmware_parse(firmware, data_ifwi_cpd, FWUPD_INSTALL_FLAG_NONE, &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+	g_assert_cmpint(fu_firmware_get_idx(firmware), ==, 0x1234);
+	data_bin = fu_firmware_write(firmware, &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(data_bin);
+	g_assert_cmpint(g_bytes_get_size(data_bin), ==, 90);
+
+	img1 = fu_firmware_get_image_by_id(firmware, "one", &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(img1);
+	g_assert_cmpint(fu_firmware_get_offset(img1), ==, 68);
+	g_assert_cmpint(fu_firmware_get_size(img1), ==, 11);
+
+	img2 = fu_firmware_get_image_by_id(firmware, "two", &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(img2);
+	g_assert_cmpint(fu_firmware_get_offset(img2), ==, 79);
+	g_assert_cmpint(fu_firmware_get_size(img2), ==, 11);
+}
+
+static void
+fu_firmware_oprom_func(void)
+{
+	gboolean ret;
+	g_autofree gchar *filename_oprom = NULL;
+	g_autoptr(FuFirmware) firmware = fu_oprom_firmware_new();
+	//	g_autoptr(FuFirmware) img1 = NULL;
+	g_autoptr(GBytes) data_bin = NULL;
+	g_autoptr(GBytes) data_oprom = NULL;
+	g_autoptr(GError) error = NULL;
+
+	filename_oprom = g_test_build_filename(G_TEST_DIST, "tests", "oprom.bin", NULL);
+	data_oprom = fu_bytes_get_contents(filename_oprom, &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(data_oprom);
+	ret = fu_firmware_parse(firmware, data_oprom, FWUPD_INSTALL_FLAG_NONE, &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+	g_assert_cmpint(fu_firmware_get_idx(firmware), ==, 0x1);
+	data_bin = fu_firmware_write(firmware, &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(data_bin);
+	g_assert_cmpint(g_bytes_get_size(data_bin), ==, 512);
+
+#if 0
+	img1 = fu_firmware_get_image_by_id(firmware, "cpd", &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(img1);
+	g_assert_cmpint(fu_firmware_get_offset(img1), ==, 68);
+	g_assert_cmpint(fu_firmware_get_size(img1), ==, 11);
+#endif
```

Comment:
why this here still

Reply:
Heh, I ran out of time and this bug still exists :) Working on it now the kids are asleep.

Reply:
Fixed, uncommented and values fixed to reflect reality.